rt/test/TESTOPR2.MLC

```diff
@@ -0,0 +1,1443 @@
+&SC2A    SETC  SIGNED(&C2A)
+         MNOTE 'C2A(''0000'')=&SC2A'
+         AIF   (&C2A NE -252645136).C2ERR
+         AIF   ('&SC2A' NE '-252645136').C2ERR
+*----------------------------------------------------------------------
+         MNOTE '********** End   C2A tests **********'
+*======================================================================
+         MNOTE '********** Begin C2B tests **********'
+*----------------------------------------------------------------------
+*        HLASM LR examples
+
+&C2B     SETC  C2B('')
+         MNOTE 'C2B('''')=>&C2B<'
+         AIF   (K'&C2B NE 0).C2ERR
+***********************************************************************
+*        TODO how to type the null character?
+*             Was able to do so on z/OS then download file.
+*             See /home/john/z390/tests/JJGOPR3.MLC.
+*             Text editor gedit complained about the 0x00 value.
+*             DO NOT ATTEMPT TO SAVE THE FILE ON A PC!!!
+*
+*             Running "bash/asm JJGOPR3 'chksrc(0)'" with next 3
+*             lines uncommented with null character in place of n
+*             yielded &C2B set to '00000000' as expected.
+***********************************************************************
+*&C2B     SETC  C2B('n')               n = null char = X'00'
+*         MNOTE 'C2B(''n'')=&C2B'
+*         AIF   ('&C2B' NE '00000000').C2ERR
+&C2B     SETC  C2B(' ')
+         MNOTE 'C2B('' '')=&C2B'
+         AIF   ('&C2B' NE '01000000').C2ERR
+&C2B     SETC  C2B('1')
+         MNOTE 'C2B(''1'')=&C2B'
+         AIF   ('&C2B' NE '11110001').C2ERR         
+&C2B     SETC  C2B('1234')
+         MNOTE 'C2B(''1234'')=&C2B'
+         AIF   ('&C2B' NE '11110001111100101111001111110100').C2ERR
+&C2B     SETC  C2B('A1')           Begin z390 tests
+         MNOTE 'C2B(''A1'')=&C2B'
+         AIF   ('&C2B' NE '1100000111110001').C2ERR
+&C2B     SETC  C2B('+')
+         MNOTE 'C2B(''+'')=&C2B'
+*        AIF   ('&C2B' NE '1100000111110001').C2ERR
+&C2B     SETC  C2B('&&')              error if only 1; returns 2
+         MNOTE 'C2B(''&&&&'')=&C2B' 
+         AIF   ('&C2B' NE '0101000001010000').C2ERR
+*z390         AIF   ('&C2B' NE '01010000').C2ERR  
+&C2B     SETC  C2B('''')              error if only 1; returns 1
+         MNOTE 'C2B('''''''')=&C2B' 
+         AIF   ('&C2B' NE '01111101').C2ERR
+*----------------------------------------------------------------------
+         MNOTE '********** End   C2B tests **********'
+*======================================================================
+         MNOTE '********** Begin C2D tests **********'
+*----------------------------------------------------------------------
+*        HLASM LR examples
+
+&C2D     SETC  C2D('')                 at most 4 characters
+         MNOTE 'C2D('''')=&C2D'
+         AIF   ('&C2D' NE '+0').C2ERR
+***********************************************************************
+*        TODO: See JJGOPR3.MLC for next test; it succeeded.
```

Comment:
Please remove todo text. Reference to items not in the repo are fine while work is in progress, but should not be needed after delivery.